src/gui/tracks/TrackContentWidget.cpp

```diff
@@ -636,13 +639,29 @@ void TrackContentWidget::paintEvent( QPaintEvent * pe )
 
 
 
+void TrackContentWidget::setMinimumWidthBasedOnClips()
+{
+	// if this is in the pattern editor
+	if (m_trackView->trackContainerView()->fixedClips())
+	{
+		if (m_clipViews.size() > 0)
+		{
+			auto minWidth = static_cast<size_t>(m_clipViews[0]->getClip()->length().getTicks());
+			setMinimumWidth(minWidth);
+			//auto minWidth = static_cast<size_t>(m_clipViews[0]->getClip()->length().nextFullBar());
+			//setMinimumWidth(minWidth * m_trackView->trackContainerView()->pixelsPerBar());
+		}
```

Comment:
I would personally rewrite it like this and remove the commented out code.

```suggestion
	if (!m_trackView->trackContainerView()->fixedClips()) { return; }
	if (m_clipViews.empty()) { return; }

	const auto minimumWidth = static_cast<size_t>(m_clipViews.first()->getClip()->length().getTicks());
	setMinimumWidth(minimumWidth);
```